src/js/application.js

```diff
@@ -0,0 +1,317 @@
+
+        [
+            {
+                dayElement: document.querySelector(selectors.TOMORROW),
+                date: new Date(weather.getForecastForDay(1).timestamp * 1000),
+            },
+            {
+                dayElement: document.querySelector(selectors.SECOND_DAY),
+                date: new Date(weather.getForecastForDay(2).timestamp * 1000),
+            },
+            {
+                dayElement: document.querySelector(selectors.THIRD_DAY),
+                date: new Date(weather.getForecastForDay(3).timestamp * 1000),
+            }
+        ].forEach(({ dayElement, date }) => {
+            dayElement.innerHTML = days[date.getDay()];
+        });
+    }
+    
+    showWeatherIcons(weather) {
+        [
+            {
+                iconElement: document.querySelector(selectors.TODAY_WEATHER_ICON),
+                weather: weather.getForecastForDay(),
+            },
+            {
+                iconElement: document.querySelector(selectors.TOMORROW_WEATHER_ICON),
+                weather: weather.getForecastForDay(1),
+            },
+            {
+                iconElement: document.querySelector(selectors.SECOND_DAY_WEATHER_ICON),
+                weather: weather.getForecastForDay(2),
+            },
+            {
+                iconElement: document.querySelector(selectors.THIRD_DAY_WEATHER_ICON),
+                weather: weather.getForecastForDay(3),
+            }
+        ].forEach(({ iconElement, weather }) => {
+            const iconFile = weatherIcons[weather.icon];
+
+            iconElement.style.background = `center no-repeat url("../assets/images/weather icons/${iconFile}")`;            
+            iconElement.style.backgroundSize = 'cover';
+        });        
+    }
+    
+    showLoadedWeather() { 
+        this.showTodaysWeatherDescription(this.currentWeather.getForecastForDay());
+        this.showTemperatureForecast(this.currentWeather);  
+        this.showNextDaysOfWeek(this.currentWeather);  
+        this.showWeatherIcons(this.currentWeather); 
+    }
+
+    async setLanguage(language, updateLocation = true) {
+        this.language = language;
+        const selectLanguageElement = document.querySelector(`[value=${this.language}]`);
+        selectLanguageElement.setAttribute('selected', 'selected');
+
+        this.translateContent();
+        if (updateLocation) {
+            const location = new Location();
+            await location.defineLocationByName(this.currentLocation.cityName, this.language);
+            this.currentLocation = location.getLocationData(); 
+            
+            const time = new Time(this.currentLocation.latitude, this.currentLocation.longitude, this.currentLocation.language);
+            await time.loadLocationTimezone();
+            const timeZone = time.getTimeZone();
+            await this.setLocation(this.currentLocation, timeZone);                        
```

Comment:
Can see this logic a few times in a file. Can it be reused by using one method?

Reply:
fixed